lib/pushmi_pullyu/cli.rb

```diff
@@ -160,15 +130,54 @@ def run_tick_loop
     end
   end
 
+  def running?
+    @running
+  end
+
+  def setup_log
+    if config.daemonize
+      PushmiPullyu::Logging.initialize_logger(config.logfile)
+    else
+      logger.formatter = PushmiPullyu::Logging::SimpleFormatter.new
+    end
+    logger.level = ::Logger::DEBUG if config.debug
+  end
+
+  def setup_signal_traps
+    Signal.trap('INT') { Thread.new { shutdown } }
+    Signal.trap('TERM') { Thread.new { shutdown } }
+    Signal.trap('HUP') { Thread.new { PushmiPullyu::Logging.reopen } }
+  end
+
+  def setup_queue
+    @queue = PushmiPullyu::PreservationQueue.new(connection: { host: config.redis_host, port: config.redis_port },
+                                                 queue_name: config.queue_name,
+                                                 age_at_least: config.minimum_age)
+  end
+
+  def shutdown
+    exit!(1) unless running?
```

Comment:
Might be worth a comment to explain the logic here, it took me a sec to figure it out.

Also, `exit!(1)` signals an error to any supervising software. Is that the desired semantics on a "forced" shutdown? If it were being run by upstart or something, double-interrupting would force it to reboot?